src/python/pants/backend/cc/lint/clangformat/rules.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import Iterable
+
+from pants.backend.cc.lint.clangformat.subsystem import ClangFormat
+from pants.backend.cc.target_types import CCSourceField
+from pants.backend.python.util_rules.pex import Pex, PexProcess, PexRequest
+from pants.core.goals.fmt import FmtRequest, FmtResult
+from pants.core.util_rules.config_files import ConfigFiles, ConfigFilesRequest
+from pants.engine.fs import Digest, MergeDigests, Snapshot
+from pants.engine.process import ProcessResult
+from pants.engine.rules import Get, MultiGet, Rule, collect_rules, rule
+from pants.engine.target import FieldSet
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+from pants.util.strutil import pluralize
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class ClangFormatFmtFieldSet(FieldSet):
+    required_fields = (CCSourceField,)
+
+    sources: CCSourceField
+
+
+class ClangFormatRequest(FmtRequest):
+    field_set_type = ClangFormatFmtFieldSet
+    name = ClangFormat.options_scope
+
+
+@rule(level=LogLevel.DEBUG)
+async def clangformat_fmt(request: ClangFormatRequest, clangformat: ClangFormat) -> FmtResult:
+    if clangformat.skip:
+        return FmtResult.skip(formatter_name=request.name)
+
+    clangformat_pex_get = Get(
+        Pex,
+        PexRequest(
+            output_filename="clangformat.pex",
+            internal_only=True,
+            requirements=clangformat.pex_requirements(),
+            interpreter_constraints=clangformat.interpreter_constraints,
+            main=clangformat.main,
+        ),
+    )
```

Comment:
Use `clangformat.to_pex_request()` instead

Reply:
Ah, yes! That's what it was. I used `to_pex_request` elsewhere (pyoxidizer probably), and when I wrote this, it felt wrong - but I couldn't figure out why.

Will update tonight.